frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -62,8 +92,63 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
   const removePlugins: boolean =
     !!consoleOperatorConfig && canPatchConsoleOperatorConfig && enabledPlugins.length > 0;
 
+  const subscriptionName = subscription?.spec?.name;
+  const subscriptionNamespace = subscription?.metadata?.namespace;
+
+  const isSubmitInProgress = React.useMemo(
+    () => operatorUninstallInProgress || operandsDeleteInProgress,
+    [operatorUninstallInProgress, operandsDeleteInProgress],
+  );
+
+  const isSubmitFinished = React.useMemo(
+    () =>
+      !isSubmitInProgress &&
+      (isDeleteAllOperandsSelected ? operandsDeleteFinished : true) &&
+      operatorUninstallFinished,
+    [
+      isSubmitInProgress,
+      isDeleteAllOperandsSelected,
+      operandsDeleteFinished,
+      operatorUninstallFinished,
+    ],
+  );
+
+  const anySubmitErrors = React.useMemo(
+    () =>
+      anyOperandDeletionErrors(operandDeletionErrors) || operatorUninstallErrorMessage.length > 0,
+    [operandDeletionErrors, operatorUninstallErrorMessage],
+  );
+
+  const [operands, operandsLoaded, operandsLoadedErrorMessage] = useOperands(
+    subscriptionName,
+    subscriptionNamespace,
+  );
+
+  const closeAndRedirect = React.useCallback(() => {
+    close();
+    // if url contains subscription name (ex: "codeready-workspaces") or installedCSV version (ex: "crwoperator.v2.9.0")
+    // redirect to ClusterServiceVersion "Installed Operators" list page,
+    // else uninstalled from "Installed Operators" list page, so do not redirect
+    if (
+      window.location.pathname.split('/').includes(subscription.metadata.name) ||
+      window.location.pathname.split('/').includes(subscription?.status?.installedCSV)
+    ) {
+      history.push(resourceListPathFromModel(ClusterServiceVersionModel, getActiveNamespace()));
+    }
+  }, [close, subscription]);
+
+  React.useEffect(() => {
+    if (isSubmitFinished && !anySubmitErrors) {
+      closeAndRedirect();
+    }
+  }, [closeAndRedirect, anySubmitErrors, isSubmitFinished]);
+
   const submit = (event: React.FormEvent<HTMLFormElement>): void => {
     event.preventDefault();
+    if (isSubmitFinished) {
+      closeAndRedirect();
```

Comment:
Hi, because if there aren't any errors we want to close and redirect (useEffect), but if there are errors we display them, then use can click on the "Ok" button (previoiusly the 'Uninstall' primary button) to 'submit' and just close the modal.

I suppose we could hide the 'Ok/Uninstall/Submit' button and just rename the Cancel button to Ok rather than reusing the primary/submit button.